src/pcre2_compile_class.c

```diff
@@ -911,14 +913,45 @@ restriction is in force). Sometimes we can just extend the original range. */
 if ((options & PCRE2_CASELESS) != 0)
   {
 #ifdef SUPPORT_UNICODE
-  if ((options & (PCRE2_UTF|PCRE2_UCP)) == 0)
+  /* UTF mode. This branch is taken if we don't support wide characters (e.g.
+  8-bit library, without UTF), but we do treat those characters as Unicode
+  (if UCP flag is set). In this case, we only need to expand the character class
+  set to include the case pairs which are in the 0-255 codepoint range. */
+  if ((options & (PCRE2_UTF|PCRE2_UCP)) != 0)
+    {
+      BOOL turkish_i = (xoptions & (PCRE2_EXTRA_TURKISH_CASING|PCRE2_EXTRA_CASELESS_RESTRICT)) ==
+        PCRE2_EXTRA_TURKISH_CASING;
+      if (start < 128)
+        {
+        uint32_t lo_end = (classbits_end < 127 ? classbits_end : 127);
+        for (c = start; c <= lo_end; c++)
+          {
+          if (turkish_i && UCD_ANY_I(c)) continue;
```

Comment:
As far as I remember, start / end should be limited to 0..255 range. How can turkish_i can affect this? You should never add char > 256 to the bitset, nor process a character > 256.

Reply:
I realized this is not allowing i -> I. Ok, then we need this extra specialization. I am sorry.

Reply:
It is pretty weird, I have to admit. The existing tests and behaviour for Turkish I in 8-bit-with-UCP-no-UTF mode are very strange, because the user is explicitly saying "bytes 0x00..0xff are codepoints U+00...U+FF", but at the same time saying that `i/İ` is a case pair, even though İ can't be provided as input! For consistency though, I think it's right as it is.